File: test_magazord_nf.py
import urllib.request
import base64
import os
import json

def load_env():
    with open('.env', 'r') as f:
        for line in f:
            if '=' in line and not line.startswith('#'):
                k, v = line.strip().split('=', 1)
                os.environ[k] = v.strip("'\"")

load_env()
user = os.environ.get('VITE_MAGAZORD_USER')
pwd = os.environ.get('VITE_MAGAZORD_PASS')
base = os.environ.get('VITE_MAGAZORD_BASE_URL')

url = f"{base}/api/v2/site/pedido/0012604710252"
auth_str = f"{user}:{pwd}"
b64_auth = base64.b64encode(auth_str.encode('utf-8')).decode('ascii')

req = urllib.request.Request(url)
req.add_header('Authorization', f'Basic {b64_auth}')
req.add_header('Accept', 'application/json')

try:
    with urllib.request.urlopen(req) as response:
        text = response.read().decode('utf-8')
        try:
            data = json.loads(text)
            d = data.get('data', data)
            def find_nf(obj, path=''):
                if isinstance(obj, dict):
                    for k, v in obj.items():
                        if any(x in k.lower() for x in ['nota', 'nf', 'fiscal']):
                            print(f"{path}.{k} = {str(v)[:150]}")
                        find_nf(v, path + '.' + k if path else k)
                elif isinstance(obj, list):
                    for i, item in enumerate(obj[:3]):
                        find_nf(item, path + f'[{i}]')
            
            find_nf(d)
            if 'arrayPedidoNota' in d:
                print('arrayPedidoNota:', json.dumps(d['arrayPedidoNota'], indent=2))
        except Exception as e:
            print("Parse err:", e)
except Exception as e:
    print("HTTP err:", e)

File: test_magazord_nf.mjs
import fetch from 'node-fetch';
import dotenv from 'dotenv';
dotenv.config();

const user = process.env.VITE_MAGAZORD_USER;
const pass = process.env.VITE_MAGAZORD_PASS;
const baseUrl = process.env.VITE_MAGAZORD_BASE_URL;

async function run() {
  const url = `${baseUrl}/api/v2/site/pedido/0012604710252`;
  const authMsg = `${user}:${pass}`;
  const encodedAuth = Buffer.from(authMsg).toString('base64');
  
  console.log(`Fetching ${url}...`);
  const res = await fetch(url, {
    headers: {
      'Authorization': `Basic ${encodedAuth}`,
      'Accept': 'application/json'
    }
  });
  
  const text = await res.text();
  console.log("Status:", res.status);
  
  try {
    const json = JSON.parse(text);
    console.log("Parsed JSON, checking keys...");
    const data = json.data || json;
    
    function find_nf(obj, path='') {
        if (typeof obj === 'object' && obj !== null) {
            for (const [k, v] of Object.entries(obj)) {
                if (/nota|nf|fiscal/i.test(k)) {
                    console.log(`  MATCH: ${path ? path+'.' : ''}${k} =`, v);
                }
                if (typeof v === 'object') {
                    find_nf(v, `${path ? path+'.' : ''}${k}`);
                }
            }
        } else if (Array.isArray(obj)) {
            for (let i = 0; i < Math.min(obj.length, 3); i++) {
                find_nf(obj[i], `${path}[${i}]`);
            }
        }
    }
    
    find_nf(data);
    if(data.arrayPedidoNota) {
       console.log("arrayPedidoNota:", JSON.stringify(data.arrayPedidoNota, null, 2));
    }
  } catch(e) {
    console.log("RAW RESP:", text.substring(0, 500));
  }
}

run();
